core/trino-main/src/main/java/io/trino/connector/CatalogConnector.java

```diff
@@ -33,22 +38,34 @@ public class CatalogConnector
     private final ConnectorServices systemConnector;
     private final Optional<CatalogProperties> catalogProperties;
     private final Catalog catalog;
+    private final LocalMemoryManager localMemoryManager;
 
     public CatalogConnector(
             CatalogHandle catalogHandle,
             ConnectorName connectorName,
             ConnectorServices catalogConnector,
             ConnectorServices informationSchemaConnector,
             ConnectorServices systemConnector,
+            LocalMemoryManager localMemoryManager,
             Optional<CatalogProperties> catalogProperties)
     {
         this.catalogHandle = requireNonNull(catalogHandle, "catalogHandle is null");
         this.connectorName = requireNonNull(connectorName, "connectorName is null");
         this.catalogConnector = requireNonNull(catalogConnector, "catalogConnector is null");
         this.informationSchemaConnector = requireNonNull(informationSchemaConnector, "informationSchemaConnector is null");
         this.systemConnector = requireNonNull(systemConnector, "systemConnector is null");
+        this.localMemoryManager = requireNonNull(localMemoryManager, "localMemoryManager is null");
         this.catalogProperties = requireNonNull(catalogProperties, "catalogProperties is null");
 
+        MemoryPool memoryPool = localMemoryManager.getMemoryPool();
+        long bytes = catalogConnector.getConnector().getExtraMemoryRequirement();
+        boolean success = memoryPool.tryReserveConnectorMemory(bytes);
+        if (!success) {
+            String info = format("Connector: %s, Allocated: %s, Delta: %s",
```

Comment:
getting reserved bytes is racy with allocation, so can be misleading at times. Maybe just report requested mem from connector?